packages/passport/sdk/src/utils/token.ts

```diff
@@ -2,22 +2,40 @@ import jwt_decode from 'jwt-decode';
 import {
   User as OidcUser,
 } from 'oidc-client-ts';
-import { IdTokenPayload } from '../types';
+import { IdTokenPayload, TokenPayload } from '../types';
 
-export function isIdTokenExpired(idToken: string | undefined): boolean {
-  if (!idToken) {
-    return false;
-  }
+function isTokenExpiredOrExpiring(token: string): boolean {
+  try {
+    // try to decode the token as access token payload or id token payload
+    const decodedToken = jwt_decode<TokenPayload | IdTokenPayload>(token);
+    const now = Math.floor(Date.now() / 1000);
+
+    // Tokens without expiration claims are invalid (security vulnerability)
+    if (!decodedToken.exp) {
+      return true;
+    }
 
-  const decodedToken = jwt_decode<IdTokenPayload>(idToken);
-  const now = Math.floor(Date.now() / 1000);
-  return decodedToken.exp < now;
+    // Check if token is expired or expiring in 30 seconds or less
+    return decodedToken.exp <= now + 30;
+  } catch (error) {
+    // If we can't decode the token, assume it's invalid
+    return true;
+  }
 }
 
-export function isTokenExpired(oidcUser: OidcUser): boolean {
-  const { id_token: idToken, expired } = oidcUser;
-  if (expired) {
+export function isAccessTokenExpiredOrExpiring(oidcUser: OidcUser): boolean {
+  const { id_token: idToken, access_token: accessToken } = oidcUser;
+
+  // Handle missing tokens - assume they need to login again
```

Comment:
```suggestion
  // Handle missing tokens - assume they need to login again
```

Returning `true` doesn't trigger a login